### PR TITLE
Add new chunking configuration

### PR DESCRIPTION
**Pull request by shafreenAnfar**
This PR,

Will be introducing two new configurations as Transfer-Encoding and Chunking for both server and client connector. 

Default value of Transfer-Encoding is chunked.

Chunking sub configuration can have three values such as auto, always and never. 

auto - outbound http message is chunked based on the message size. For instance, if the message size is smaller than the configured chunk size, http message is sent with content-length and if the message size is larger than the configured chunk size, message is chunked. 

always - outbound http message is always chunked.

never - outbound http message is always sent with content-length. 
  
In addition to that, I have refactored the code a bit.

### CODE REVIEW

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/Constants.java [hunk 1]
```diff
@@ -161,9 +167,10 @@
     public static final String SSL_ENABLED_PROTOCOLS = "sslEnabledProtocols";
     public static final int OPTIONS_STRUCT_INDEX = 0;
     public static final int ENDPOINT_TIMEOUT_STRUCT_INDEX = 1;
-    public static final int ENABLE_CHUNKING_INDEX = 0;
-    public static final int IS_KEEP_ALIVE_INDEX = 1;
-    public static final int HTTP_VERSION_STRUCT_INDEX = 0;
+    public static final int TRANSFER_ENCODING = 0;
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/java/org/ballerinalang/net/http/Constants.java]: Unused constant?

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/Constants.java [hunk 2]
```diff
@@ -58,6 +58,10 @@
     public static final String HTTP_DEFAULT_HOST = "0.0.0.0";
     public static final String TLS_STORE_TYPE = "tlsStoreType";
     public static final String PKCS_STORE_TYPE = "PKCS12";
+    public static final String CHUNKED = "chunked";
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/java/org/ballerinalang/net/http/Constants.java]: Unused constant?

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java [hunk 3]
```diff
@@ -827,44 +844,90 @@ private static String getHeaderFromStruct(BStruct struct, String key) {
...
+            if (keepAliveAttrVal != null) {
+                listenerConfiguration.setKeepAlive(keepAliveAttrVal.getBooleanValue());
+            } else {
+                listenerConfiguration.setKeepAlive(Boolean.TRUE);
+            }
+
+            // For the moment we don't have to pass it down to transport as we only support
+            // chunking. Once we start supporting gzip, deflate, etc, we need to parse down the config.
+            if (transferEncoding != null) {
+                if (!"chunking".equalsIgnoreCase(transferEncoding.getStringValue())) {
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java]: Shall we use the constant ANN_CONFIG_ATTR_CHUNKING for here?

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java [hunk 4]
```diff
@@ -936,62 +984,46 @@ private static String getListenerInterface(Map<String, String> parameters) {
...
             }
             if (trustStorePasswordAttrVal != null) {
-                httpsPropMap
-                        .put(Constants.ANN_CONFIG_ATTR_TRUST_STORE_PASS, trustStorePasswordAttrVal.getStringValue());
+                listenerConfiguration.setTrustStorePass(trustStorePasswordAttrVal.getStringValue());
             }
-            if (sslEnabledProtocolsAttrVal != null) {
-                httpsPropMap.put(Constants.ANN_CONFIG_ATTR_SSL_ENABLED_PROTOCOLS,
+
+            List<Parameter> serverParams = new ArrayList();
```
Comment (anupama-pathirage) [inline modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java]: Why not `new ArrayList<>(); `here?

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/Constants.java (hunk 1 above)
Comment (shafreenAnfar) [reply modules/ballerina-http/src/main/java/org/ballerinalang/net/http/Constants.java]: This is used now :)

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/Constants.java (hunk 2 above)
Comment (shafreenAnfar) [reply modules/ballerina-http/src/main/java/org/ballerinalang/net/http/Constants.java]: Removed. There are many though. Will remove them iteratively.

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java (hunk 4 above)
Comment (shafreenAnfar) [reply modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java]: Changed..

**Pull request merged**